exonum/src/events/codec.rs

```diff
@@ -147,6 +163,39 @@ mod test {
         assert!(responder.decode(&mut bytes).is_err());
     }
 
+    #[test]
+    #[should_panic(expected = "Received malicious message with wrong total_len")]
+    fn decode_message_wrong_length() {
+        let data = vec![0u8, 0, 0, 0, 0, 0, 32, 0, 0, 0];
+        let mut bytes: BytesMut = BytesMut::new();
+        let (ref mut responder, ref mut initiator) = create_encrypted_codecs();
+        let raw = RawMessage::new(MessageBuffer::from_vec(data.clone()));
+        initiator.encode(raw, &mut bytes).unwrap();
+
+        responder.decode(&mut bytes).unwrap();
+    }
+
+    #[test]
+    fn decode_message_eof() {
+        let data = vec![0u8, 0, 0, 0, 0, 0, 10, 0, 0, 0];
+        let mut bytes: BytesMut = BytesMut::new();
+        let (ref mut responder, ref mut initiator) = create_encrypted_codecs();
+        let raw = RawMessage::new(MessageBuffer::from_vec(data.clone()));
+        initiator.encode(raw, &mut bytes).unwrap();
+
+        match responder.decode_eof(&mut bytes.clone()) {
+            Ok(Some(ref r)) if r == &RawMessage::new(MessageBuffer::from_vec(data.clone())) => {}
```

Comment:
Nit: If I understand correctly, you can use `AsRef<[u8]>` implementation for `RawMessage` here:

```rust
Ok(Some(ref message)) if *message.as_ref() == data => {}
```